packages/compass-indexes/src/components/create-search-index-modal/index.tsx

```diff
@@ -0,0 +1,186 @@
+import { useTrackOnChange } from '@mongodb-js/compass-logging';
+import {
+  Modal,
+  ModalFooter,
+  ModalHeader,
+  ModalBody,
+  TextInput,
+  Label,
+  spacing,
+  css,
+  HorizontalRule,
+  Subtitle,
+  Button,
+  Link,
+  Icon,
+  WarningSummary,
+  ErrorSummary,
+} from '@mongodb-js/compass-components';
+import { closeModal, saveIndex } from '../../modules/search-indexes';
+import { connect } from 'react-redux';
+import { CodemirrorMultilineEditor } from '@mongodb-js/compass-editor';
+import type { RootState } from '../../modules';
+import _parseShellBSON, { ParseMode } from 'ejson-shell-parser';
+import type { Document } from 'mongodb';
+
+// Copied from packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/utils.ts
+function parseShellBSON(source: string): Document[] {
+  const parsed = _parseShellBSON(source, { mode: ParseMode.Loose });
+  if (!parsed || typeof parsed !== 'object') {
+    // XXX(COMPASS-5689): We've hit the condition in
+    // https://github.com/mongodb-js/ejson-shell-parser/blob/c9c0145ababae52536ccd2244ac2ad01a4bbdef3/src/index.ts#L36
+    throw new Error('The provided index definition is invalid.');
+  }
+  return parsed;
+}
+
+export const DEFAULT_INDEX_DEFINITION = `{
+  "mappings": {
+    "dynamic": true
+  }
+}`;
+
+const bodyGapStyles = css({
+  marginTop: spacing[3],
+});
+
+const toolbarStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+});
+
+type CreateSearchIndexModalProps = {
+  isModalOpen: boolean;
+  isBusy: boolean;
+  error?: string;
+  saveIndex: (indexName: string, indexDefinition: Document) => void;
+  closeModal: () => void;
+};
+
+export const CreateSearchIndexModal: React.FunctionComponent<
+  CreateSearchIndexModalProps
+> = ({ isModalOpen, isBusy, error, saveIndex, closeModal }) => {
+  const [indexName, setIndexName] = useState<string>('default');
+  const [indexDefinition, setIndexDefinition] = useState<string>(
+    DEFAULT_INDEX_DEFINITION
+  );
+  const [parsingError, setParsingError] = useState<string | undefined>(
+    undefined
+  );
+
+  const onSearchIndexDefinitionChanged = useCallback(
+    (newDefinition: string) => {
+      setParsingError(undefined);
+
+      try {
+        parseShellBSON(newDefinition);
+        setIndexDefinition(newDefinition);
+      } catch (ex) {
+        setParsingError((ex as Error).message);
+      }
+    },
+    [setIndexDefinition, setParsingError]
+  );
+
+  const onSaveIndex = useCallback(() => {
+    if (parsingError) {
+      return;
+    }
+
+    const indexDefinitionDoc = parseShellBSON(indexDefinition);
+    saveIndex(indexName, indexDefinitionDoc);
+  }, [saveIndex, parsingError, indexName, indexDefinition]);
+
+  useTrackOnChange(
+    'COMPASS-SEARCH-INDEXES-UI',
+    (track) => {
+      if (isModalOpen) {
+        track('Screen', { name: 'create_search_index_modal' });
+      }
+    },
+    [isModalOpen],
+    undefined,
+    React
+  );
+
+  return (
+    <Modal
```

Comment:
I seem to remember that we have FormModal component in shared components package, maybe something we can use here?

Reply:
Let me check, I've seen it somewhere, I think we can make it work.

Reply:
The styles are different in FormModal (no separator, spacing is different) and adapting it would be tweaking the FormModal. I'm not sure it is worth it, as it's used in a few places already.